test/e2e/e2e_suite_test.go

```diff
@@ -158,9 +160,26 @@ func initScheme() *runtime.Scheme {
 func loadE2EConfig(configPath string) *clusterctl.E2EConfig {
 	config := clusterctl.LoadE2EConfig(context.TODO(), clusterctl.LoadE2EConfigInput{ConfigPath: configPath})
 	Expect(config).ToNot(BeNil(), "Failed to load E2E config from %s", configPath)
+	Expect(config.Variables).To(HaveKey(CNIPath))
+	// Read CNI file and set CNI_RESOURCES environmental variable
+	cniManifestPath, ok := config.Variables[CNIPath]
```

Comment:
What about moving this code into  `clusterctl.LoadE2EConfig` so everyone using the test framework gets this behavior by default?

Reply:
Moving this makes sense if we also move `setCNIResources` to `clusterctl.LoadE2EConfig`. 
Otherwise all can be done here so that we do not define CNIPath in the framework. WDYT?

Reply:
I think that it is better to do this in the framework vs copy-pasting the same changes in all the places where the framework is used.
So I'm ok in moving `setCNIResources` to the framework as well.

Reply:
On second thought, if we move CNI check to the framework, then we assume providers will always use a single CNI for all their tests. This was not the behaviour before, each test could pick a different CNI manifest file during `ApplyClusterTemplateAndWait`. 

I only moved `setCNIResources ` to the framework and renamed it as `SetCNIEnvVar`. By this way, if same CNI will be used across CNIs, it can be called from `loadE2EConfig` in  e2e_suite_test, otherwise, both `CNI` and `CNI_RESOURCES` can be set differently for each test and `SetCNIEnvVar` can be called once during each Spec.